src/storage/lsm.rs

```diff
@@ -132,7 +132,8 @@ impl LSM {
     }
 
     pub fn put(&mut self, k: Key, v: Value) -> Result<()> {
-        serde::serialize_kv(&k, &v, &mut self.commit_log)?;
+        k.ser(&mut self.commit_log)?;
```

Comment:
it might simplify some call-sites to expose a function in `serde` like this:

``` rust
serde::serialize_kv(k, v, &mut file) -> Result<int> {
  i = k.ser(file)?;
  i += v.ser(file)?;
  i
}
```